packages/next/src/server/app-render/app-render.tsx

```diff
@@ -1364,6 +1368,7 @@ async function renderToHTMLOrFlightImpl(
     requestStore,
     serverActions,
     ctx,
+    temporaryReferences,
```

Comment:
probably should just move the temp reference creation into handleAction

Reply:
maybe handleAction should return the temporary reference set and we only pass it to the `renderStream` function if it exists. that way we don't use these references unless they're part of a response for an action?

Reply:
That makes way more sense.  I just threw them at everything since I wasn't sure how these parts fit together.

Reply:
yeah basically the only time you'd need to render these references is if there was a paired action handled by decodeReply and you wanted to be able to respond with the reference itself. So action pathway needs it but regular renders for both SSR and RSC don't really need it since their is no pair with which to create these references